src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcKey.cs

```diff
@@ -25,7 +25,53 @@ internal static partial class Crypto
         [return: MarshalAs(UnmanagedType.Bool)]
         internal static extern bool EcKeyUpRef(IntPtr r);
 
-        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyGetCurveName")]
-        internal static extern int EcKeyGetCurveName(SafeEcKeyHandle ecKey);
+        [DllImport(Libraries.CryptoNative)]
+        private static extern int CryptoNative_EcKeyGetSize(SafeEcKeyHandle ecKey, out int keySize);
+        internal static int EcKeyGetSize(SafeEcKeyHandle key)
+        {
+            int keySize;
+            int rc = CryptoNative_EcKeyGetSize(key, out keySize);
+            if (rc == 1)
+            {
+                return keySize;
+            }
+            throw Interop.Crypto.CreateOpenSslCryptographicException();
+        }
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern int CryptoNative_EcKeyGetCurveName(SafeEcKeyHandle ecKey, out int nid);
+
+        internal static string EcKeyGetCurveName(SafeEcKeyHandle key)
+        {
+            int nidCurveName;
+            int rc = CryptoNative_EcKeyGetCurveName(key, out nidCurveName);
+            if (rc == 1)
+            {
+                if (nidCurveName == Interop.Crypto.NID_undef)
+                {
+                    Debug.Assert(false); // Key is invalid or doesn't have a curve
```

Comment:
FWIW, I think we should be provided messages in most/all of our Debug.Asserts/Fails in code paths that run on Unix, anyway.  Since stack traces currently lack line number information, it's much easier to determine which assert failed if it has a meaningful message, plus in a case like this, ideally you'd want to include the value of ```nidCurveName``` in the message.